### DISCUSSION TITLE
Issue with enum type

### DISCUSSION THREAD

**Discussion by rehmaninvozone** [prisma/prisma#16900] [Q&A] [answered]
Hi, I am getting an issue with the enum type this may not be a Prisma issue. 
My schema is something like this where startOfWeek is an enum.
```
model Company{
  id String @id @default(auto()) @map("_id") @db.ObjectId
  name String @unique
  settings Setting?
  startOfWeek WeekStart? @default(MONDAY)
  @@map("companies")
}

enum WeekStart {
  SUNDAY
  MONDAY
}
```

company entity is like this

```
import { Field, ObjectType, registerEnumType } from '@nestjs/graphql';
import { WeekStart } from '@prisma/client';
import { BaseModel } from '../../common/models/base.model';

registerEnumType(WeekStart, {
  name: 'WeekStart',
  description: 'week start',
});

@ObjectType()
export class Company extends BaseModel {
  name: string;
  @Field(() => WeekStart, { defaultValue: 'MONDAY' })
  startOfWeek?: WeekStart;
}
```

The issue is I am getting this error
```Error: Cannot determine a GraphQL input type ("Object") for the "equals". Make sure your class is decorated with an appropriate decorator.```

I have enabled the graphql cli plugin so, I don't need to specify return types but when I remove the ```@Field(() => WeekStart, { defaultValue: 'MONDAY' })``` I am getting this error
``` Cannot determine a GraphQL output type for the "startOfWeek". Make sure your class is decorated with an appropriate decorator.```

I don't know from where this error is coming. I even did the same thing for input.ts file. Hope anyone can help with this. @nurul3101

**Comment by nurul3101** [comment, marked as answer]
Hi @rehmaninvozone,

Instead of passing 'MONDAY' as a string in the defaultValue attribute of the `@Field` decorator, could you try passing the enum `WeekStart.MONDAY` instead?

```typescript
@ObjectType()
export class Company extends BaseModel {
  name: string;
  @Field(() => WeekStart, { defaultValue: WeekStart.MONDAY })
  startOfWeek?: WeekStart;
}
```

**Reply by rehmaninvozone** [reply]
getting same error ```Error: Cannot determine a GraphQL input type ("Object") for the "equals". Make sure your class is decorated with an appropriate decorator.``` anything else can I try?

**Reply by nurul3101** [reply]
Did you start getting this error after updating any dependencies?

**Reply by rehmaninvozone** [reply]
Oh sorry, forgot to update you. It was fixed. Even though I am using the graphQL CLI plugin not to add `@filed()` but still somehow Nest or GraphQl not able to detect the output of the enum type. So, I have to use `@Field(() => WeekStart)` and the issue was coming from other files. Thanks for your help.

**Reply by nurul3101** [reply]
I am glad to hear that you were able to resolve it! 😄

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
